Clean up WebKitLegacy's LayerFlushController/WebViewLayerFlushScheduler

smfr opened the pull request:
#### 66783d616f3d0e3fd47f2a05e5ae7387ba5f5732
<pre>
Clean up WebKitLegacy&apos;s LayerFlushController/WebViewLayerFlushScheduler
<a href="https://bugs.webkit.org/show_bug.cgi?id=249826">https://bugs.webkit.org/show_bug.cgi?id=249826</a>
rdar://103653788

Reviewed by Tim Horton.

In WebKitLegacy, both LayerFlushController and WebViewLayerFlushScheduler existed, where
LayerFlushController owned a WebViewLayerFlushScheduler. These classes can be combined.

Confusingly, LayerFlushController was mostly implemented in WebViewData.mm, but LayerFlushController::flushLayers()
was in WebView.mm, making it hard to find.

So combine the classes into WebViewRenderingUpdateScheduler which lives in its own file. This class
does not need to be RefCounted. WebView methods that it calls are declared in WebViewInternal.h.

An iBooks-specific quirk is removed since Books no longer uses UIWebView.

* Source/WebKitLegacy/SourcesCocoa.txt:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _close]):
(-[WebView close]):
(-[WebView _scheduleUpdateRendering]):
(-[WebView _updateRendering]):
(-[WebView _willStartRenderingUpdateDisplay]):
(-[WebView _didCompleteRenderingUpdateDisplay]):
(LayerFlushController::flushLayers): Deleted.
* Source/WebKitLegacy/mac/WebView/WebViewData.h:
(): Deleted.
(LayerFlushController::create): Deleted.
(LayerFlushController::didCompleteRenderingUpdateDisplay): Deleted.
* Source/WebKitLegacy/mac/WebView/WebViewData.mm:
(currentRunLoop): Deleted.
(LayerFlushController::scheduleLayerFlush): Deleted.
(LayerFlushController::invalidate): Deleted.
(LayerFlushController::LayerFlushController): Deleted.
(WebViewLayerFlushScheduler::WebViewLayerFlushScheduler): Deleted.
(WebViewLayerFlushScheduler::~WebViewLayerFlushScheduler): Deleted.
(WebViewLayerFlushScheduler::schedule): Deleted.
(WebViewLayerFlushScheduler::invalidate): Deleted.
(WebViewLayerFlushScheduler::layerFlushCallback): Deleted.
* Source/WebKitLegacy/mac/WebView/WebViewInternal.h:
* Source/WebKitLegacy/mac/WebView/WebViewRenderingUpdateScheduler.h: Added.
* Source/WebKitLegacy/mac/WebView/WebViewRenderingUpdateScheduler.mm: Added.
(currentRunLoop):
(WebViewRenderingUpdateScheduler::WebViewRenderingUpdateScheduler):
(WebViewRenderingUpdateScheduler::scheduleRenderingUpdate):
(WebViewRenderingUpdateScheduler::invalidate):
(WebViewRenderingUpdateScheduler::didCompleteRenderingUpdateDisplay):
(WebViewRenderingUpdateScheduler::registerCACommitHandlers):
(WebViewRenderingUpdateScheduler::renderingUpdateRunLoopObserverCallback):
(WebViewRenderingUpdateScheduler::updateRendering):

Canonical link: <a href="https://commits.webkit.org/258283@main">https://commits.webkit.org/258283@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/26f4426055b88986bd456c67262e0171e37798ae

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/101450 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/10607 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/34504 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/110720 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/170975 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/11559 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/1461 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/93836 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/108537 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107231 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/8794 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/92035 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/35318 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/90691 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/23443 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/78317 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4209 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/24954 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4272 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/1375 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10358 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44443 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5695 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6034 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebKitLegacy/mac/WebView/WebView.mm:
```diff
@@ -8836,88 +8808,45 @@ - (void)_setNeedsOneShotDrawingSynchronization:(BOOL)needsSynchronization
+- (void)_updateRendering
+{
+#if PLATFORM(IOS_FAMILY)
+    // Ensure fixed positions layers are where they should be.
```
Perhaps fix the grammar-o while you move this?

hortont424 commented on Source/WebKitLegacy/mac/WebView/WebViewRenderingUpdateScheduler.mm:
```diff
@@ -0,0 +1,173 @@
+    // we still allow this, see <rdar://problem/7403328>. Since the race condition and subsequent
+    // crash are especially troublesome for iBooks, we never allow the observer to be added to the
+    // main run loop in iBooks.
+    if (WebCore::CocoaApplication::isIBooks())
```
Books doesn't use legacy WebKit anymore, maybe time to remove this?

hortont424 commented on Source/WebKitLegacy/mac/WebView/WebViewRenderingUpdateScheduler.mm:
```diff
@@ -0,0 +1,173 @@
+}
+
+/*
+    Note: Much of the following is osbsolete.
```
"osbsolete"

smfr committed:
Clean up WebKitLegacy's LayerFlushController/WebViewLayerFlushScheduler
https://bugs.webkit.org/show_bug.cgi?id=249826
rdar://103653788

Reviewed by Tim Horton.

In WebKitLegacy, both LayerFlushController and WebViewLayerFlushScheduler existed, where
LayerFlushController owned a WebViewLayerFlushScheduler. These classes can be combined.

Confusingly, LayerFlushController was mostly implemented in WebViewData.mm, but LayerFlushController::flushLayers()
was in WebView.mm, making it hard to find.

So combine the classes into WebViewRenderingUpdateScheduler which lives in its own file. This class
does not need to be RefCounted. WebView methods that it calls are declared in WebViewInternal.h.

An iBooks-specific quirk is removed since Books no longer uses UIWebView.

* Source/WebKitLegacy/SourcesCocoa.txt:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _close]):
(-[WebView close]):
(-[WebView _scheduleUpdateRendering]):
(-[WebView _updateRendering]):
(-[WebView _willStartRenderingUpdateDisplay]):
(-[WebView _didCompleteRenderingUpdateDisplay]):
(LayerFlushController::flushLayers): Deleted.
* Source/WebKitLegacy/mac/WebView/WebViewData.h:
(): Deleted.
(LayerFlushController::create): Deleted.
(LayerFlushController::didCompleteRenderingUpdateDisplay): Deleted.
* Source/WebKitLegacy/mac/WebView/WebViewData.mm:
(currentRunLoop): Deleted.
(LayerFlushController::scheduleLayerFlush): Deleted.
(LayerFlushController::invalidate): Deleted.
(LayerFlushController::LayerFlushController): Deleted.
(WebViewLayerFlushScheduler::WebViewLayerFlushScheduler): Deleted.
(WebViewLayerFlushScheduler::~WebViewLayerFlushScheduler): Deleted.
(WebViewLayerFlushScheduler::schedule): Deleted.
(WebViewLayerFlushScheduler::invalidate): Deleted.
(WebViewLayerFlushScheduler::layerFlushCallback): Deleted.
* Source/WebKitLegacy/mac/WebView/WebViewInternal.h:
* Source/WebKitLegacy/mac/WebView/WebViewRenderingUpdateScheduler.h: Added.
* Source/WebKitLegacy/mac/WebView/WebViewRenderingUpdateScheduler.mm: Added.
(currentRunLoop):
(WebViewRenderingUpdateScheduler::WebViewRenderingUpdateScheduler):
(WebViewRenderingUpdateScheduler::scheduleRenderingUpdate):
(WebViewRenderingUpdateScheduler::invalidate):
(WebViewRenderingUpdateScheduler::didCompleteRenderingUpdateDisplay):
(WebViewRenderingUpdateScheduler::registerCACommitHandlers):
(WebViewRenderingUpdateScheduler::renderingUpdateRunLoopObserverCallback):
(WebViewRenderingUpdateScheduler::updateRendering):

Canonical link: https://commits.webkit.org/258283@main
